src/scripts/main.js

```diff
@@ -1,3 +1,248 @@
 'use strict';
 
-// write your code here
+let valueBoard;
+const rows = 4;
+const columns = 4;
+let score = 0;
+
+const buttonStart = document.querySelector('.start');
+const fieldRows = document.querySelectorAll('tr');
+const cells = document.querySelectorAll('td');
+const messageStart = document.querySelector('.message-start');
+const messageWin = document.querySelector('.message-win');
+const messageLose = document.querySelector('.message-lose');
+
+const board = [];
+
+[...fieldRows].forEach(row => {
+  const cell = row.querySelectorAll('td');
+
+  board.push(cell);
+});
+
+function setGame() {
+  valueBoard = [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
+
+  addNumber();
+  addNumber();
+
+  for (let i = 0; i < rows; i++) {
+    for (let j = 0; j < rows; j++) {
+      const number = valueBoard[i][j];
+
+      updateCell(board[i][j], number);
+    }
+  }
+}
+
+function updateCell(cell, num) {
+  cell.classList.value = `field-cell`;
+
+  if (num > 0) {
+    cell.textContent = num;
+    cell.classList.add(`field-cell--${num}`);
+  } else if (num === 0) {
+    cell.textContent = '';
+  }
+}
+
+function createRandomNumber(num) {
+  return Math.floor(Math.random() * num);
+}
+
+function addNumber() {
+  if (!hasEmptyCell()) {
+    return;
+  }
+
+  let activeCell = false;
+
+  while (!activeCell) {
+    const r = createRandomNumber(rows);
+    const c = createRandomNumber(columns);
+
+    if (valueBoard[r][c] === 0) {
+      const num = Math.random() <= 0.1 ? 4 : 2;
+
+      valueBoard[r][c] = num;
+      board[r][c].textContent = num;
+      board[r][c].classList.add(`field-cell--${num}`);
+      activeCell = true;
+    }
+  }
+};
+
+function hasEmptyCell() {
+  for (let i = 0; i < rows; i++) {
+    for (let j = 0; j < columns; j++) {
+      if (valueBoard[i][j] === 0) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+};
+
+document.addEventListener('click', (ev) => {
+  if (ev.target.closest('.start')) {
+    messageStart.classList.add(`hidden`);
+    buttonStart.textContent = 'Restart';
+    buttonStart.classList.add(`restart`);
+    buttonStart.classList.remove(`start`);
+    setGame();
+
+    document.addEventListener('keydown', (e) => {
+      switch (e.key) {
+        case 'ArrowRight':
+          moveRight();
+          addNumber();
+          break;
+        case 'ArrowLeft':
+          moveLeft();
+          addNumber();
+          break;
+        case 'ArrowDown':
+          moveDown();
+          addNumber();
+          break;
+        case 'ArrowUp':
+          moveUp();
+          addNumber();
+          break;
+        default:
+          return;
+      }
+
+      [...cells].forEach(cell => {
+        if (cell.textContent === '2048') {
+          messageWin.classList.remove(`hidden`);
+
+          setTimeout(() => {
+            document.location.reload();
```

Comment:
_not critical_. not sure that the desired behavior - you may remove the event listener to prevent the next moves, etc
up to you whether to change it or not